File: data_tree/Node.java

public class Node {
    //node for a binary tree
	private String data;
	private Node left;
	private Node right;
	
	public Node(String data, Node left, Node right){
		this.data = data;
		this.left = left;
		this.right = right;
	} // end constructor
	
	public void setData(String data){
		this.data = data;
	}
	
	public String getData(){
		return this.data;
	}
	
	public void setLeft(Node left){
		this.left = left;
	}
	public Node getLeft(){
		return this.left;
	}
	
	public void setRight(Node right){
		this.right = right;
	}
	public Node getRight(){
		return this.right;
	}	
}

File: data_tree/Node.cpp
#include "Node.h"


Node::Node(){
  Node::data = "";
  Node::left = NULL;
  Node::right = NULL;
}

Node::Node(std::string data, Node * left, Node * right){
  Node::data = data;
  Node::left = left;
  Node::right = right;
}

void Node::setData(std::string data){
  Node::data = data;
}
void Node::setLeft(Node * left){
  Node::left = left;
}
void Node::setRight(Node * right){
  Node::right = right;
}
std::string Node::getData(){
  return Node::data;
}
Node * Node::getLeft(){
  return Node::left;
}
Node * Node::getRight(){
  return Node::right;
}
